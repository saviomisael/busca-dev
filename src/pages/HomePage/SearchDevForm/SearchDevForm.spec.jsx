import { screen, fireEvent } from '@testing-library/react';
import { waitFor } from '@testing-library/react';
import { SearchDevForm } from '.';

describe('<SearchDevForm />', () => {
  it('should calls onSubmit prop when form is submit', async () => {
    const onSubmitMock = jest.fn();

    global.renderTheme(<SearchDevForm onSubmit={onSubmitMock} />);

    const buttonSubmit = screen.getByRole('button');

    fireEvent.submit(buttonSubmit);

    await waitFor(() => {
      expect(buttonSubmit).toBeDisabled();
    });

    expect(onSubmitMock).toBeCalledTimes(1);
  });

  it('should matches with snapshot', () => {
    const onSubmitMock = jest.fn();

    const { container } = global.renderTheme(<SearchDevForm onSubmit={onSubmitMock} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
