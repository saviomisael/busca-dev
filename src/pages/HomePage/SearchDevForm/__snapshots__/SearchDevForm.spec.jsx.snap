// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchDevForm /> should matches with snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: none;
  border-radius: 0.6rem;
  width: 100%;
  background: #D9D9D9;
}

.c0 > * {
  background-color: transparent;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem 1rem;
  border: none;
  font-size: 1.4rem;
  color: #353535;
}

.c1::-webkit-input-placeholder {
  color: #353535CC;
  font-size: 1.4rem;
}

.c1::-moz-placeholder {
  color: #353535CC;
  font-size: 1.4rem;
}

.c1:-ms-input-placeholder {
  color: #353535CC;
  font-size: 1.4rem;
}

.c1::placeholder {
  color: #353535CC;
  font-size: 1.4rem;
}

.c2 {
  border-top-right-radius: inherit;
  border-bottom-right-radius: inherit;
}

.c2:hover,
.c2:active {
  background: #D9D9D9;
}

.c2:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

.c2:active {
  -webkit-filter: brightness(1.08);
  filter: brightness(1.08);
}

.c2:disabled {
  cursor: not-allowed;
}

.c3 {
  -webkit-filter: invert(16%) sepia(1%) saturate(0%) hue-rotate(332deg) brightness(102%) contrast(86%);
  filter: invert(16%) sepia(1%) saturate(0%) hue-rotate(332deg) brightness(102%) contrast(86%);
}

<form
  class="c0"
>
  <input
    class="c1"
    name="username"
    placeholder="Digite o username do github"
  />
  <button
    class="c2"
    type="submit"
  >
    <img
      class="c3"
      src="search_icon.svg"
    />
  </button>
</form>
`;
